File: Labs/04 Annalyzer/analyzer.py
# Jose Reynaldo and Marie Klinizing
import stringdata
import time

def main():
    cont = stringdata.get_data() #pull data from stringdata.py
    checks=["not_here", "mzzzz","aaaaa"] #create list for checks

    for i in range(len(checks)): #iterates through check list
        start=time.time() #print out the info from the linear search

        print("Testing Linear Search for:", checks[i])
        index=linear_search(cont, checks[i])
        end=time.time()-start

        if index==-1:
            print(checks[i], "not found")
        else:
            print(checks[i], "found at", index)
        print("Linear Search Time:", end, "\n")


        start = time.time() #print out the checks for the binary search
        print("Testing Binary Search for:", checks[i])

        index = binary_search(cont, checks[i])
        end = time.time() - start


        if index == -1:
            print(checks[i], "not found")
        else:
            print(checks[i], "found at", index)
        print("Binary Search Time: ", end, "\n")

def linear_search(container, element) -> int:
    for i, val in enumerate(container): #checks each element
        if val==element:
            return i
    return -1

def binary_search(container, element) -> int:
    min=0 #define min, max, mid
    max=len(container)-1
    while min <= max:
        mid = min + (max - min) // 2
        if element<container[mid]: #if not equal to mid, redefine max or min
            max=mid-1
        elif element>container[mid]:
            min=mid+1
        else:
            return mid
    return -1

if __name__=="__main__":
    main() #call main

File: Labs/lab 5/analyzer.cpp
//
// Created by jose reynaldo and Karan Faldu
//

#include <iostream>
#include <ctime>
#include "StringData.h"

using namespace std;
int linearSearch(std::vector<std::string> container, std::string element);
int binarySearch(std::vector<std::string> container, std::string element);

int main()
{

    std::vector<std::string> cont = getStringData();
    string checks[3] = {"not_here", "mzzzz","aaaaa"};

    int counter = 3;
    for (int i = 0; i < counter; i++)
    {
        float start = time(NULL);
        cout << "Testing Linear Search for:" << checks[i] ;
        int index = linearSearch(cont, checks[i]);
        float end = time(NULL);
        if (index== -1)
        {
            cout << checks[i] << "not found";
        }
        else
        {
            cout << checks[i] << "found at" << index;
        }
        cout << "Linear Search Time:" << end;

        start = time(NULL);
        cout << "Testing Binary Search for:" << checks[i];
        index = binarySearch(cont, checks[i]);
        end = time(NULL);

        if (index == -1) {
            cout << checks[i] << "not found";
        } else {
            cout << checks[i] << "found at" << index;
        }
        cout << "Binary Search Time:" << end;
    }

    return 0;
}

int linearSearch(std::vector<std::string> container, std::string element) {
    for(int i = 0; i< container.size(); i++){
        if (container[i] == element){
            return i;
        }
    }
    return -1;
}

int binarySearch(std::vector<std::string> container, std::string element)
{
    int min = 0;
    int max = container.size() - 1;
    int mid = min + (max - min) / 2;
    while (min <= max)
    {
        mid = min + (max - min) / 2;
        if (element<container[mid])
        {
            max=mid-1;
        }
        else if (element > container[mid])
        {
            min=mid+1;
        }
        else
        {
            return mid;
        }
    }
    return -1;
}